packages/devtools_app/test/trees_test.dart

```diff
@@ -94,27 +94,35 @@ void main() {
       expect(parent.children.first, equals(child));
       expect(child.parent, equals(parent));
     });
-  });
 
-  test('containsChildWithCondition', () {
-    expect(
-      treeNode0.containsChildWithCondition((TestTreeNode node) {
-        return node == treeNode1;
-      }),
-      isTrue,
-    );
-    expect(
-      treeNode0.containsChildWithCondition((TestTreeNode node) {
-        return node.children.length == 2;
-      }),
-      isTrue,
-    );
-    expect(
-      treeNode0.containsChildWithCondition((TestTreeNode node) {
-        return node.isExpanded;
-      }),
-      isFalse,
-    );
+    test('containsChildWithCondition', () {
+      expect(
+        treeNode0.containsChildWithCondition((TestTreeNode node) {
+          return node == treeNode1;
+        }),
+        isTrue,
+      );
+      expect(
+        treeNode0.containsChildWithCondition((TestTreeNode node) {
+          return node.children.length == 2;
+        }),
+        isTrue,
+      );
+      expect(
+        treeNode0.containsChildWithCondition((TestTreeNode node) {
+          return node.isExpanded;
+        }),
+        isFalse,
+      );
+    });
+
+    test('firstNodeAtLevel', () {
```

Comment:
nit: add a todo to add more tests for this method. For example, test for when the level requested is too large.